core/src/main/java/org/apache/calcite/rel/metadata/RelMdUtil.java

```diff
@@ -872,23 +872,36 @@ public static boolean checkInputForCollationAndLimit(RelMetadataQuery mq,
       RelNode input, RelCollation collation, RexNode offset, RexNode fetch) {
     // Check if the input is already sorted
     boolean alreadySorted = collation.getFieldCollations().isEmpty();
-    for (RelCollation inputCollation : mq.collations(input)) {
-      if (inputCollation.satisfies(collation)) {
-        alreadySorted = true;
-        break;
+    if (!alreadySorted) {
+      final ImmutableList<RelCollation> collations = mq.collations(input);
+      if (collations != null) {
```

Comment:
this is the NPE fix

Reply:
Should it be like `if (collations == null) { return false; }` ?

Reply:
Yes, totally. I'll change it

Reply:
Changed